webui/src/pages/repositories/services/import_data.jsx

```diff
@@ -0,0 +1,198 @@
+        importStatusUpdate.numObj = sum
+        updateImportState(importStatusUpdate);
+    } while (paginationResp.has_more);
+    const metarange = await metaRanges.createMetaRange(repoId, rangeArr);
+
+    try {
+        importBranchResp = await branches.get(repoId, branch);
+    } catch (error) {
+        if (error instanceof NotFoundError) {
+            importBranchResp = await createBranch(repoId, refId, branch);
+        } else {
+            throw error;
+        }
+    }
+    await commits.commit(repoId, importBranchResp.id, commitMsg, {}, metarange.id);
+    importStatusUpdate.importPhase = ImportPhase.Completed;
+    updateImportState(importStatusUpdate);
+}
+
+const createBranch = async (repoId, refId, branch) => {
+    // Find root commit for repository
+    let hasMore = true;
+    let nextOffset = "";
+    let baseCommit = refId;
+    do {
+        let response = await commits.log(repoId, refId, nextOffset, 1000);
+        hasMore = response.pagination.has_more;
+        nextOffset = response.pagination.next_offset;
+        baseCommit = response.results.at(-1);
+    } while (hasMore)
+    await branches.create(repoId, branch, baseCommit.id);
+    return await branches.get(repoId, branch);
+}
+
+const ImportProgress = ({numObjects}) => {
+    return (<Row>
+        <Col>
+            <div className='import-text'>
+                <p>Imported <strong>
+                    <div className='import-num-objects'> {numObjects} </div>
+                </strong> objects so far...
+                </p>
+            </div>
+            <div>
+                <LinearProgress color="success"/>
+            </div>
+            <div>
+                <small><abbr><br/>Please leave this tab open while import is in progress...</abbr></small>
+            </div>
+        </Col>
+    </Row>);
+}
+
+const ImportDone = ({numObjects, importBranch, currBranch = ''}) => {
+    return (<Row>
+        <Col>
+            <div className={"mt-10 mb-2 mr-2 row mt-4 import-success"}>
+                <p><strong>Success!</strong></p>
+            </div>
+            <div className='import-text'>
+                <p><strong>
+                    <div className='import-num-objects'> {numObjects} </div>
+                </strong> objects imported and committed into branch {importBranch}.
+                </p>
+            </div>
+            {(currBranch && importBranch !== currBranch) &&
+                <div className='import-text'>
+                    <p> Use the&nbsp;<Link to={`compare?ref=${currBranch}&compare=${importBranch}`}
+                                           variant="success">Compare tab</Link>&nbsp;to view the changes and merge
+                        them to {currBranch}.
+                    </p>
+                </div>
+            }
+        </Col>
+    </Row>);
+}
+const ExecuteImportButton = ({isEnabled, importPhase, importFunc}) => {
+    return (
+        <Button variant="success"
+                disabled={!isEnabled}
```

Comment:
You need to update the state when the import is in progress and completed to disable the import button. Currently, you can press it during and at the of the import and get an error.

Reply:
Thanks!!!